### ISSUE TITLE
PhysicsDirectSpaceState3D.intersect_ray return type in C# not implemented

### ISSUE THREAD

**Issue by plinio-jrm** [godotengine/godot-docs#8354]
## **Godot version:** 
4.1.1.stable.mono.official

## **Issue description:**
In GDScript, we can access any value. 
example: `result.position`, 
However in C# there is no type implemented that allow us to access the method return values.

In Visual Studio, when navigating to the definition we have this information:
```cs
public Dictionary IntersectRay(PhysicsRayQueryParameters3D parameters) {
    return NativeCalls.godot_icall_1_729(MethodBind1, GodotObject.GetPtr(this), GodotObject.GetPtr(parameters));
}
```
Which give us the value from a memory pointer. It is not possible to cast or convert anything to access the values, for example if i want the "position". The `result.position` in C# is not acceptable by the compiler.

### Return Suggestion
```cs
public struct IntersectRayReturn {
   public Vector3 Position;
   public Vector3 Normal;
   public int ColliderId;
   public CollisionObject3D Collider;
   public int Shape;
   public Rid Rid;
}
```

### TLDR
Implement a return type in C# for `PhysicsDirectSpaceState3D.intersect_ray` to access it's values in their respective type.

## **URL to the documentation page:**
https://docs.godotengine.org/en/stable/classes/class_physicsdirectspacestate3d.html#class-physicsdirectspacestate3d-method-intersect-ray

**Comment by AThousandShips** [comment]
You can access the results, see [here](https://docs.godotengine.org/en/stable/tutorials/scripting/c_sharp/c_sharp_collections.html#doc-c-sharp-collections-dictionary) on how to use `Dictionary`

**Comment by plinio-jrm** [comment]
Thank you so much. 
now i understand.
